### PR TITLE
Update badges documentation

### PR DESCRIPTION
**Pull request by EdorianDark**
After removal of [badges](https://github.com/rust-lang/crates.io/issues/2436) in crates.io  the documentation should be updated accordingly.

### CODE REVIEW

File: src/doc/src/guide/continuous-integration.md [hunk 1]
```diff
@@ -49,6 +49,40 @@ breakage in nightly will not fail your overall build. Please see the
 [GitLab CI](https://docs.gitlab.com/ce/ci/yaml/README.html) for more
 information.
 
+### Github Actions
```
Comment (ehuss) [inline src/doc/src/guide/continuous-integration.md:52]: This change seems unrelated?

**Comment by ehuss** [comment]
https://github.com/rust-lang/crates.io/issues/2436#issuecomment-620495491 seems to imply that it is not yet deprecated. Is it maybe premature to remove these? FWIW, I don't mind either way, it just seems like this hasn't been set in stone, yet.  AFAIK, lib.rs doesn't show these, but do other registries?

**Comment by EdorianDark** [comment]
The documentation stated, that these badges would be displayed in crates.io.
But they are not displayed  any more and the cited comment notes, there is not much reason to keep these fields.

**Comment by ehuss** [comment]
ping @rust-lang/crates-io are you OK with removing the CI badges documentation at this time, which will show up in the stable docs in 1.45 (July 16)?

File: src/doc/src/guide/continuous-integration.md (hunk 1 above)
Comment (EdorianDark) [reply src/doc/src/guide/continuous-integration.md:52]: Oh indeed. Is it ok here or is a seperate pull request better?

Comment (ehuss) [reply src/doc/src/guide/continuous-integration.md:52]: I think it's probably fine to be here.

File: src/doc/src/guide/continuous-integration.md [hunk 2]
```diff
@@ -49,6 +49,40 @@ breakage in nightly will not fail your overall build. Please see the
...
+
+To test your package on Github Actions, here is a sample `workflow.yml` file, which has to be in the directory `.github/workflows directory`:
```
Comment (ehuss) [inline src/doc/src/guide/continuous-integration.md:54]: It would probably be best to use a capital `H` in the name "GitHub" throughout.

```suggestion
To test your package on GitHub Actions, here is a sample `workflow.yml` file, which has to be in the directory `.github/workflows`:
```

**Comment by jtgeibel** [comment]
The crates.io team discuss this in our meeting today, and there were no objections to this change and targeting the 1.45 release. I didn't review the GH Actions additions in this PR, but the removal of all badges with the exception of maintenance status looks great. Thanks!

File: src/doc/src/reference/manifest.md [hunk 3]
```diff
@@ -433,59 +433,11 @@ default-run = "a"
...
-  Pipelines.
-- Available badges pertaining to code test coverage are Codecov and Coveralls.
-- There are also maintenance-related badges based on isitmaintained.com
-  which state the issue resolution time, percent of open issues, and future
-  maintenance intentions.
-
-Most badge specifications require a `repository` key. It is expected to be in
-`user/repo` format.
+[crates.io] does not display various badges for build status, test coverage, etc. for
+each crate any more. Only the following fields will be shown.
```
Comment (ehuss) [inline src/doc/src/reference/manifest.md:437]: I would not start off with what badges *don't* do.  Perhaps start off with a description, such as:

```
Badges are small status pieces of status information about a package. They can
be displayed as small images on a package registry website. 
```

Then follow with a description of the badges that are supported by crates.io.  I would keep the description of the maintenance badges, and the link to isitmaintained.com, for example.

Then, at the bottom after the example, include a note for anyone who may be confused as to what changed:

> Note: [crates.io] previously displayed badges next to a crate on its
> website, but that functionality has been removed. Packages should place
> badges in its README file and set [the readme field](#the-readme-field) so
> that the entire README is displayed on [crates.io].

Also, I'm uncertain, but it looks like crates.io is not showing *any* badges anymore?  What is the state of the maintenance badges?  Why would someone set them?

Also, the format for the fields is not explained.  What is the form for the `repository` field? Is it a URL to a website? A git URL?

Comment (EdorianDark) [reply src/doc/src/reference/manifest.md:437]: Since the [maintenance status is moved into the gui](https://github.com/rust-lang/crates.io/issues/2437) maybe the whole section should be removed?

**Comment by behnam** [comment]
Looks like the problem with the current changes comes from the fact that there's actually nothing left for Cargo to do with badge attributes in the manifest.

I think we need to either clarify their reason for existence, or just put the in the "deprecated and not used anymore" category. The latter would require changing the code, as well, to raise a warning when the attribute is used.

Anyone knows if there's an issue already open for this?

**Comment by ehuss** [comment]
I think it's fine to wait until there is a decision on how to handle the maintenance status.  That is tracked in https://github.com/rust-lang/crates.io/issues/2437 and https://github.com/rust-lang/crates.io/issues/2438.

**Comment by ehuss** [comment]
I'm going to close this now that #8727 is merged.  The `maintenance` badge is left in the docs under the anticipation that it may still be used in the future, though it seems unlikely and that may also get removed later.  Thanks for the PR, and sorry about the uncertainty and moving slowly on this.
